### PR TITLE
Task 2:  decorate class

### PR DESCRIPTION
**Pull request by Metsanpeitto**
#### In this task, I have implemented the different features:

- Create a class `Corrector` with a method `correct_name` that does the following:
  - The first letter of the given name is a capital letter.
  - The name has a maximum of 10 characters. If it's longer it should trim the word.
- Set an instance of `Corrector` in `Person` on initialization.
- In `Person` add a method `validate_name` that uses the `Corrector` instance to validate and save the corrected name.

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Corrector
+  def correct_name(name)
+    first_letter = name[0].capitalize
+    str = name
+    str[0] = first_letter
+    str[0, 11]
```
Comment (arslanbisharat) [inline corrector.rb]: - [ ] It is requested in the project instructions to make sure that the name has a maximum of 10 characters but it's 11 for you. Let's trim the word,
Capture

File: person.rb [hunk 2]
```diff
@@ -13,6 +15,11 @@ def can_use_services?
     of_age? || parent_permission == true
```
Comment (arslanbisharat) [inline person.rb:16]: - [ ] Let's add an instance of `Corrector` set in the `Person` class on initialization following this [lesson](
`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/composition_by_examples.md#composable-classes`). 
Capture

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: corrector.rb (hunk 1 above)
Comment (Metsanpeitto) [reply corrector.rb]: :heavy_check_mark: Fixed!

File: person.rb (hunk 2 above)
Comment (Metsanpeitto) [reply person.rb:16]: :+1:  Done!

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉 

## Hi @Metsanpeitto, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**